src/lib/proof_system.ts

```diff
@@ -427,21 +434,34 @@ function ZkProgram<
       compile,
       verify,
       digest,
+      analyzeMethods,
       publicInputType: publicInputType as ProvableOrUndefined<
         Get<StatementType, 'publicInput'>
       >,
       publicOutputType: publicOutputType as ProvableOrVoid<
         Get<StatementType, 'publicOutput'>
       >,
-      analyzeMethods,
+      privateInputTypes: Object.fromEntries(
```

Comment:
exposing the private input types on zkprogram seems consistent, since we already expose the public input types and the methods

Reply:
What is this exactly for?

Reply:
It's in support of a testing utility I added: https://github.com/o1-labs/o1js/pull/1220#discussion_r1392513182
You can ignore it, it's not related to the circuit gadgets added here :D